### ISSUE TITLE
Only first test finshes, chromedriver doesn't stop

### ISSUE THREAD

**Issue by rostockahoi** [laravel/dusk#341]
`chromedriver-linux` spawns when running `artisan dusk` but it doesn't stop after test execution, so every subsequent test will never finish because there cannot be multiple instances of chromedriver on the same port.

When I try to exec chromedriver manually it says:
`[0,002][SEVERE]: bind() returned an error, errno=98: Die Adresse wird bereits verwendet` which means "the address is already in use".

However if I kill `chromedriver-linux` manually it works again - once.

How can I fix the missing stop instruction?

**Comment by calebporzio** [comment]
@rostockahoi - in your `TestCase.php` make sure your `prepare` method has the `@beforeClass` annotation like so:

```php
/**
     * Prepare for Dusk test execution.
     *
     * @beforeClass
     * @return void
     */
    public static function prepare()
    {
        static::startChromeDriver();
    }
```

Let me know if that doesn't fix your issue.

**Comment by rostockahoi** [comment]
Hi @calebporzio, thanks for the reply. The line exists. That's why the first test run works. But as soon as there's a process chromderiver-linux that is not terminated after the test has finished the next chromedriver-linux instance runs into an error.

Something regarding the chromedriver-linux termination seems to not work properly.

Btw: I didn't change Dusk's code.

**Comment by rostockahoi** [comment]
I think I've come closer to the problem. It seems everything is working just fine in `vendor/laravel/dusk/src/SupportsChrome.php`. It starts and stops the `chromedriver-linux` process as expected. However, on my machine (Linux Mint 18.2) there are spawning 2 processes of which only one gets terminated after the test has finished. The other one remains running and is then blocking port `9515`. The blocked port is preventing the next test to be finshed because `chromedriver-linux` then errors with the line seen in my first post above.

I did a screencast of the processes running a single test which works. Then again which doesn't work due to the blocking process. Then killing this process manually and running the test again successfully.

![dusk 1](https://user-images.githubusercontent.com/16897125/29718736-08413f78-89b4-11e7-9fa5-c80a6b5d8950.gif)

**Comment by raniesantos** [comment]
I have the exact same problem. I need to keep killing the process to make dusk work again.

**Comment by glatzor** [comment]
ProcessBuilder doesn't use the exec command to run the chromedriver. You should switch to pure Process. See https://github.com/symfony/symfony/issues/5759 for some background information.

I worte a small Trait to fix the issue in my project - you have to adapt the namespace and relative path to chromedriver:

```
<?php

namespace Tests\Helper\Traits;yYyy

use RuntimeException;
use Symfony\Component\Process\Process;

trait FixSupportsChrome
{
    /**
     * Build the process to run the Chromedriver.
     *
     * @throws \RuntimeException if the driver file path doesn't exist.
     *
     * @return \Symfony\Component\Process\Process
     */
    protected static function buildChromeProcess()
    {
        $driver = static::$chromeDriver
                ?: realpath(__DIR__.'/../../../vendor/laravel/dusk/bin/chromedriver-'.static::driverSuffix());

        if (realpath($driver) === false) {
            throw new RuntimeException("Invalid path to Chromedriver [{$driver}].");
        }

        return new Process([realpath($driver)], null, static::chromeEnvironment());
    }
}
```

**Comment by pratikid** [comment]
I had a similar issue in Windows. @rostockahoi 
After executing command `php artisan dusk pathToTestFile.php`

![capture](https://user-images.githubusercontent.com/10076073/30277527-b3edcdf4-9725-11e7-92a0-bb2aaa34dad9.JPG)

It opened the browser, executed the test case but didn't terminate the test case in the command prompt. So, I got to know from @rostockahoi post that my **chromedriver-win.exe** in `vendor\laravel\dusk\bin\` wasn't getting terminated automatically.

Then I checked the `path` in My Computer->Properties->Advanced system settings->Environment Variables->Edit in User Variables and added **C:\Windows\System32**

Because there is a "TaskKill.exe" in this path.

![capture](https://user-images.githubusercontent.com/10076073/30277922-daf6a424-9726-11e7-85ea-545514eec4b8.JPG)

And after opening new command prompt
Finally it worked !!! .

![capture](https://user-images.githubusercontent.com/10076073/30278548-3e0d8ec8-9728-11e7-8978-5bc5a13cc61f.JPG)

I think some similar way must be there in linux to solve the issue.
